test/compress/reduce_vars.js

```diff
@@ -53,9 +53,7 @@ reduce_vars: {
             console.log(-3);
             eval("console.log(a);");
         })(eval);
-        (function() {
-            return "yes";
-        })();
+        "yes";
```

Comment:
`drop_unused()` used to indiscriminiately returns `AST_EmptyStatement` if everything within a `var` statement is dropped.

But if you are `in_list` (since we are within `AST_Function.body`) you can return `MAP.skip` instead. So that causes the next `descend()` to [compress](https://github.com/mishoo/UglifyJS2/blob/88e7a542cd8a8406f54c53cefe72944452f5e013/lib/compress.js#L3012-L3018) in `OPT(AST_Call)`